lib/livebook/apps/teams_app_spec.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Livebook.Apps.TeamsAppSpec do
+  # App spec for organization apps fetched from Livebook Teams.
+
+  @enforce_keys [:slug, :version, :hub_id, :app_deployment_id]
+
+  defstruct [:slug, :version, :hub_id, :app_deployment_id]
+end
+
+defimpl Livebook.Apps.AppSpec, for: Livebook.Apps.TeamsAppSpec do
+  alias Livebook.Hubs
+
+  def load(app_spec, files_tmp_path) do
+    with {:ok, hub} <- fetch_hub(app_spec.hub_id),
+         {:ok, app_deployment, derived_key} <-
+           fetch_download_info(hub.id, app_spec.app_deployment_id),
```

Comment:
So the idea is that we store as little information as possible, just the hub id and app deployment id, then we fetch everything as needed to make the request.